src/pybind/mgr/volumes/fs/fs_util.py

```diff
@@ -75,6 +75,27 @@ def listdir(fs, dirpath):
         raise VolumeException(-e.args[0], e.args[1])
     return dirs
 
+def listsnaps(fs, snapdirpath, filter_inherited_snaps=False):
+    """
+    Get the snap names from a given .snap directory path
+    """
+    if os.path.basename(snapdirpath) != b".snap":
+        raise VolumeException(-errno.EINVAL, "Not a snap directory: {0}".format(snapdirpath))
+    snaps = []
+    try:
+        with fs.opendir(snapdirpath) as dir_handle:
+            d = fs.readdir(dir_handle)
+            while d:
+                if (d.d_name not in (b".", b"..")) and d.is_dir():
+                    if not d.d_name.startswith(b"_"):
```

Comment:
Do ancestor snapshots always begin with '_' or is the prefix configurable?

Reply:
I don't think so. @batrick ?

Reply:
Always I believe.

Reply:
maybe you could use inherited_snap(d.d_name) here?